api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -54,6 +58,7 @@
   private transient Map<Integer, String> idToName = null;
 
   public Schema(List<NestedField> columns, Map<String, Integer> aliases) {
+    this.schemaId = 0;
```

Comment:
I think we should add a constant for the default schema ID. That way we can reference it in other places when we fill in the default.